examples/pykokkos/streams.py

```diff
@@ -0,0 +1,27 @@
+import pykokkos as pk
+
+import cupy as cp
+
+@pk.workunit
+def print_stream(i, x):
+    if x == 0:
+        pk.printf("Default stream\n")
+    elif x == 1:
+        pk.printf("Stream 1\n")
+    elif x == 2:
+        pk.printf("Stream 2\n")
+
+if __name__ == "__main__":
+    space = pk.Cuda
+
+    s1 = cp.cuda.Stream()
+    s2 = cp.cuda.Stream()
+
+    instance1 = pk.ExecutionSpaceInstance(space, s1)
```

Comment:
ExecutionSpaceInstance => ExecutionSpace

Reply:
if we are instantiating a stream on top of a space, we could also go with `space.make(stream)`. (And potentially have also a version that does not accept a stream but uses a new stream for each call.)

Reply:
That does seem like a better interface. I'll address this in a new PR